### PR TITLE
Cleanup plugin api surface

### PR DESCRIPTION
**Pull request by ilya-g**

### CODE REVIEW

File: main/src/kotlinx/team/infra/InfraExtension.kt [hunk 1]
```diff
@@ -26,12 +26,18 @@ open class InfraExtension(val project: Project) {
...
     private var apiCheckHandler: ((ApiCheckConfiguration) -> Unit)? = null
-    internal fun afterApiCheck(handler: (ApiCheckConfiguration) -> Unit) {
+    @Suppress("DEPRECATION")
+internal fun afterApiCheck(handler: (ApiCheckConfiguration) -> Unit) {
         apiCheckHandler = handler
     }
 
+    @Deprecated("No longer supported. Use binary-compatibility-validator instead.")
+    @Suppress("DEPRECATION")
     fun apiCheck(configure: Action<ApiCheckConfiguration>) {
```
Comment (ilya-g) [inline main/src/kotlinx/team/infra/InfraExtension.kt]: Perhaps we could drop this functionality immediately if we can make sure nobody uses it.

Comment (qurbonzoda) [reply main/src/kotlinx/team/infra/InfraExtension.kt]: I am not aware of any library that uses it.

File: main/src/kotlinx/team/infra/InfraExtension.kt [hunk 2]
```diff
@@ -26,12 +26,18 @@ open class InfraExtension(val project: Project) {
...
     }
 
+    @Deprecated("No longer supported. Use binary-compatibility-validator instead.")
+    @Suppress("DEPRECATION")
     val apiCheck = project.objects.newInstance<ApiCheckConfiguration>()
+    @Suppress("DEPRECATION")
     private var apiCheckHandler: ((ApiCheckConfiguration) -> Unit)? = null
-    internal fun afterApiCheck(handler: (ApiCheckConfiguration) -> Unit) {
+    @Suppress("DEPRECATION")
+internal fun afterApiCheck(handler: (ApiCheckConfiguration) -> Unit) {
```
Comment (qurbonzoda) [inline main/src/kotlinx/team/infra/InfraExtension.kt]: This has shifted left

File: main/src/kotlinx/team/infra/Utils.kt [hunk 3]
```diff
@@ -19,20 +19,19 @@ fun cleanup(file: File) {
     }
 }
 
-inline fun <reified T : Task> Project.task(
+internal inline fun <reified T : Task> Project.task(
     name: String,
-    depends: String? = null,
+    depends: String? = null, // TODO: safe delete parameter
```
Comment (qurbonzoda) [inline main/src/kotlinx/team/infra/Utils.kt]: Should this TODO be addressed within this PR?

File: main/src/kotlinx/team/infra/Utils.kt [hunk 4]
```diff
@@ -42,10 +41,10 @@ class KotlinClosure1<T : Any?, V : Any>(
     fun doCall(it: T): V? = it.function()
 }
 
-fun <T> Any.closureOf(action: T.() -> Unit) : Closure<T> =
+internal fun <T> Any.closureOf(action: T.() -> Unit) : Closure<T> =
```
Comment (qurbonzoda) [inline main/src/kotlinx/team/infra/Utils.kt]: I can't find any usages of `closureOf` and `KotlinClosure1` within the infra project. Is it safe to remove them?

File: main/src/kotlinx/team/infra/InfraExtension.kt (hunk 1 above)
Comment (qurbonzoda) [reply main/src/kotlinx/team/infra/InfraExtension.kt]: But I don't mind having it deprecated for some time.

File: main/src/kotlinx/team/infra/Utils.kt (hunk 3 above)
Comment (ilya-g) [reply main/src/kotlinx/team/infra/Utils.kt]: I thought about preserving compatibility, but seems that it's not required

File: main/src/kotlinx/team/infra/Utils.kt [hunk 5]
```diff
@@ -1,13 +1,12 @@
...
 
-import groovy.lang.*
 import org.gradle.api.*
 import org.gradle.api.logging.*
 import org.gradle.api.model.ObjectFactory
 import org.gradle.api.tasks.*
 import java.io.*
 
-fun cleanup(file: File) {
+internal fun cleanup(file: File) {
```
Comment (qurbonzoda) [inline main/src/kotlinx/team/infra/Utils.kt]: This function seems to be redundant now. It was previously used by ApiCheck tasks.

**Review by qurbonzoda — APPROVED** [review]

File: main/src/kotlinx/team/infra/Utils.kt (hunk 5 above)
Comment (ilya-g) [reply main/src/kotlinx/team/infra/Utils.kt]: Good eye!

**Pull request merged**